query_optimizer/tests/ExecutionHeuristics_unittest.cpp

```diff
@@ -75,12 +75,12 @@ class ExecutionHeuristicsTest : public ::testing::Test {
     std::vector<attribute_id> build_attribute_ids;
     build_attribute_ids.push_back(build_attribute_id);
     QueryPlan::DAGNodeIndex build_operator_index =
-        query_plan->addRelationalOperator(new BuildHashOperator(*build_relation,
+        query_plan->addRelationalOperator(new BuildHashOperator(0,  /* dummy query ID */
```

Comment:
It is up to you to take the following suggestion.

In each unit test, we could have a const member called `query_id`, and pass it to each operator and its work orders.

And it is better to explicitly use a variable to represent the same query id value shared by all the operators, instead of multiple query id `0`s.